### PR TITLE
Change suggestion of field when not in self context

### LINKED ISSUE #60057: Nonsensical extra diagnostics for E0425 in static methods
The following code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=dae0528ebd48ed0e5552b6f5e374de7a)):

```rust
struct A {
    banana: u8,
}

impl A {
    fn new(peach: u8) -> A {
        A {
            banana: banana // note: banana not in scope here
        }
    }
}
```

produces the following error:

```
error[E0425]: cannot find value `banana` in this scope
 --> src/lib.rs:8:21
  |
8 |             banana: banana
  |                     ^^^^^^
  |                     |
  |                     `self` value is a keyword only available in methods with `self` parameter
  |                     help: try: `self.banana`
```

First, `help` is wrong, because this is a static method and self is not available in scope. Second it is not clear why a line above `help` is talking about `self` at all.

### PR DESCRIPTION
**Pull request by estebank**
Fix #60057.

### CODE REVIEW

**Comment by estebank** [comment]
CC @nagisa

**Review by davidtwco — APPROVED** [review]
LGTM

File: src/test/ui/issues/issue-60057.stderr [hunk 1]
```diff
@@ -0,0 +1,20 @@
+error[E0425]: cannot find value `banana` in this scope
+  --> $DIR/issue-60057.rs:8:21
+   |
+LL |             banana: banana
+   |                     ^^^^^^ a field by this name exists in `Self::banana`
```
Comment (petrochenkov) [inline src/test/ui/issues/issue-60057.stderr]:
```suggestion
   |                     ^^^^^^ a field by this name exists in `Self`
```
A field exists in a field?

File: src/test/ui/class-missing-self.stderr [hunk 2]
```diff
@@ -2,7 +2,7 @@ error[E0425]: cannot find value `meows` in this scope
   --> $DIR/class-missing-self.rs:9:7
    |
 LL |       meows += 1;
-   |       ^^^^^ help: try: `self.meows`
+   |       ^^^^^ help: you might have meant to use the available field: `self.meows`
```
Comment (petrochenkov) [inline src/test/ui/class-missing-self.stderr]: (As the author of that "try", I still prefer less wordy labels.)

Comment (estebank) [reply src/test/ui/class-missing-self.stderr]: The longer text is there for people that come from languages like Java, where implicit `this`. Having the longer text let us differentiate between fields and bindings when communicating with users.

File: src/test/ui/issues/issue-60057.stderr (hunk 1 above)
Comment (estebank) [reply src/test/ui/issues/issue-60057.stderr]: Done.

**Comment by Centril** [comment]
Failed in https://github.com/rust-lang/rust/pull/60100#issuecomment-484740120, @bors r-

**Pull request merged**